### PR TITLE
GeoParquet Reader Implemented (SEDONA-94)

### PR DESCRIPTION
**Pull request by ashar236**
## Is this PR related to a proposed Issue?
SEDONA-94 
## What changes were proposed in this PR?
GeoParquet reader support was added.
## How was this patch tested?
The unit test for three different parquet file with geometry column have been added.
## Did this PR include necessary documentation updates?
Yes.

### CODE REVIEW

**Comment by MulesNose** [comment]
Great stuff, found your question from Opengeospatial and almost linked you to your own commit!

**Comment by cholmes** [comment]
Is there any progress on this? I'd love to add Sedona to the list at https://github.com/opengeospatial/geoparquet#current-implementations--examples

**Comment by JassonHua** [comment]
您好，您的邮件我已收到，我会尽快给您回复。谢谢!

**Comment by jiayuasu** [comment]
> Is there any progress on this? I'd love to add Sedona to the list at [opengeospatial/geoparquet#current-implementations--examples](https://github.com/opengeospatial/geoparquet#current-implementations--examples)

We are still working on this. The PR will be updated soon. Will keep you posted

File: sql/src/main/scala/org/apache/spark/sql/sedona_sql/io/geoparquet/ParquetFileFormat.scala [hunk 1]
```diff
@@ -0,0 +1,545 @@
+/*
```
Comment (ericsun95) [inline sql/src/main/scala/org/apache/spark/sql/sedona_sql/io/geoparquet/ParquetFileFormat.scala:1]: Looks like all these are duplicated with Spark internal parquet reader implementation. From my understanding the geoparquet is just one specific type of parquet with its only schema. 
Generally, what differs here? how about implementing a very lean layer on top of spark native parquet file reader using DataSourceV2 interface with specific schema?

Comment (neontty) [reply sql/src/main/scala/org/apache/spark/sql/sedona_sql/io/geoparquet/ParquetFileFormat.scala:1]: @jiayuasu if I start working on geoparquet predicate pushdown, will it be invalidated if you are moving to the datasource V2 api soon? Is there a branch with the v2 api ("[We have a neat version that nicely supports Spark 3.3 only](https://github.com/apache/incubator-sedona/pull/652)")?

Comment (jiayuasu) [reply sql/src/main/scala/org/apache/spark/sql/sedona_sql/io/geoparquet/ParquetFileFormat.scala:1]: @neontty My understanding is that the merged PR in Sedona already uses Datasource v2 as Spark Parquet reader is using it. But Spark parquet reader is designed in a way that most critical functions are private, so we have to copy lots of their code for adding a few changes on our own.

The version that supports Spark 3.3 only is here: https://github.com/ashar236/incubator-sedona/tree/geoparquet-spark3.3/sql/src/main/scala/org/apache/spark/sql/execution/datasources/parquet

In short, I believe your geoparquet predicate pushdown function will Not be invalidated. But I would suggest that please minimize your change to the codebase.

**Comment by jiayuasu** [comment]
> Is there any progress on this? I'd love to add Sedona to the list at [opengeospatial/geoparquet#current-implementations--examples](https://github.com/opengeospatial/geoparquet#current-implementations--examples)

@cholmes Dear Chris, we have released Sedona 1.3.0 which includes reader/writer for GeoParquet. I created [a PR](https://github.com/opengeospatial/geoparquet/pull/150) on geoparquet repo as well. Please review it when you have time :-)
